### ISSUE TITLE
[FR]

### ISSUE THREAD

**Issue by JJSong0805** [mlflow/mlflow#2748]
Thank you for submitting a feature request. **Before proceeding, please review MLflow's
[Issue Policy for feature requests](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md#feature-requests)
and the [MLflow Contributing Guide](https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst)**.

**Please fill in this feature request template to ensure a timely and thorough response.**

## Willingness to contribute
The MLflow Community encourages new feature contributions. Would you or another member of your organization be willing to contribute an implementation of this feature (either as an MLflow Plugin or an enhancement to the MLflow code base)?

- [ ] Yes. I can contribute this feature independently.
- [O] Yes. I would be willing to contribute this feature with guidance from the MLflow community.
- [ ] No. I cannot contribute this feature at this time.

## Proposal Summary
We wonder does it make sense to have a dynamic sorting function at model comparison interface?
For now, we can only sort models by column before choosing the model to compare. 

## Motivation
- What is the use case for this feature?
For visualization of model metrics comparison.
- Why is this use case valuable to support for MLflow users in general?
The function should improve the experience of users who wants to sort the model results by metrics on the figure.
- Why is this use case valuable to support for your project(s) or organization?
We are users who wants to sort the model results by metrics on the figures, and that improves our experience :)
- Why is it currently difficult to achieve this use case? (please be as specific as possible about why related MLflow features and components are insufficient)
For now, we can only sort models by column before choosing the model to compare.  If we want to sort the models by other metrics we need to go back to do the same thing every time.

## Which MLflow component(s) does this feature affect?

- [O] UI
- [ ] Command Line Interface
- [ ] API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Model Registry
- [ ] Scoring
- [ ] R
- [ ] Java
- [ ] Python

## Details
NAN
(Use this section to include any additional information about the feature. If you have a proposal
for how to implement this feature, please include it here. For implementation guidelines, please
refer to the [Contributing Guide](https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst#contribution-guidelines).)

**Comment by tomasatdatabricks** [comment]
This makes sense to me. cc @Zangr

**Comment by Zangr** [comment]
This makes sense. I'd recommend implementing a grid view similar to the new runs table and use it both in run comparison and model comparison pages. CC @gioa for design.

![Screen Shot 2020-04-30 at 6 46 20 AM](https://user-images.githubusercontent.com/6811562/80718159-a5461e80-8aae-11ea-81e5-b779fd392e68.png)

**Comment by gioa** [comment]
@JJSong0805 Thanks for the idea! Just want to clarify, is this what you are asking for - sort by a metric and dynamically update the columns on the right side? (in the mock it is a run comparison page, but can work similarly in the model version compare page) 
![sortOnComparison](https://user-images.githubusercontent.com/1960721/80853050-8052c800-8be2-11ea-983a-7828d9f12f26.png)

@Zangr As we talked offline, the comparison page's sorting header is on the left column, not sure whether the table component you suggested supports that.

**Comment by JJSong0805** [comment]
@gioa Yes! That's it! Thanks to all of you and look forward to the update :).

**Comment by Zangr** [comment]
@gioa Thank you for providing the design. @JJSong0805 you are welcome to submit a PR for this feature. Building on top of the current simple table view should be good. We can always convert to a virtualized table solution later.
